postings.go

```diff
@@ -158,24 +154,31 @@ func Merge(its ...Postings) Postings {
 }
 
 type mergePostings struct {
-	a, b     Postings
-	aok, bok bool
-	cur      uint32
+	a, b        Postings
+	initialized bool
+	aok, bok    bool
+	cur         uint32
 }
 
 func newMergePostings(a, b Postings) *mergePostings {
-	it := &mergePostings{a: a, b: b}
-	it.aok = it.a.Next()
-	it.bok = it.b.Next()
-
-	return it
+	return &mergePostings{a: a, b: b, aok: true, bok: true}
 }
 
 func (it *mergePostings) At() uint32 {
 	return it.cur
 }
 
 func (it *mergePostings) Next() bool {
+	if !it.initialized {
+		if !it.a.Next() {
+			it.aok = false
+		}
+		if !it.b.Next() {
+			it.bok = false
+		}
+		it.initialized = true
+	}
```

Comment:
> it seems to just make things more compliciated by adding initialized and checking it on every iteration.

Yes, unfortunately Go does not support coroutines. (In Python, it would be as simple as [this](https://gist.github.com/telendt/3fe7f3db9301d40cb3291c1d05428c21)).

The alternative to flag checking on each iteration is to swap function pointer swap. Would you prefer that?

> Is there any thing fundamentally speaking the one element peeking?
> Also we should at least be able to simplify the ifs to it.aok = it.a.Next() like in the former constructor, right?

I'm not sure I understand. Are you asking what are the disadvantages of advancing `it.(a|b)` in the constructor? If so, I already gave some example to @Gouthamve.